azure/services/privatedns/privatedns.go

```diff
@@ -59,7 +61,15 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	if zoneSpec != nil {
 		// Create the private DNS zone.
 		s.Scope.V(2).Info("creating private DNS zone", "private dns zone", zoneSpec.ZoneName)
-		err := s.client.CreateOrUpdateZone(ctx, s.Scope.ResourceGroup(), zoneSpec.ZoneName, privatedns.PrivateZone{Location: to.StringPtr(azure.Global)})
+		pDNS := privatedns.PrivateZone{
+			Location: to.StringPtr(azure.Global),
+			Tags: converters.TagsToMap(infrav1.Build(infrav1.BuildParams{
+				ClusterName: s.Scope.ClusterName(),
+				Lifecycle:   infrav1.ResourceLifecycleOwned,
+				Additional:  s.Scope.AdditionalTags(),
+			})),
+		}
+		err := s.client.CreateOrUpdateZone(ctx, s.Scope.ResourceGroup(), zoneSpec.ZoneName, pDNS)
 		if err != nil {
 			return errors.Wrapf(err, "failed to create private DNS zone %s", zoneSpec.ZoneName)
 		}
```

Comment:
we should consider also not creating/deleting vnet links if both the DNS zone and the vnet are not managed by capz

Reply:
I was thinking, simply we should not be reconciling/deleting vnet links if the vnet is not managed. Does the management of vnet depends on wether DNS is managed or not?

Reply:
By the way, I have raised the commit according to above comment of mine. Let me know if this is incorrect or needs changes.